ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift

```diff
@@ -0,0 +1,123 @@
+                _ = viewController.view
+            }
+            
+            it("should have a correct view model type") {
+                expect(viewController.viewModel).to(beAKindOf(AccountAddressFormViewModel.self))
+            }
+            
+            it("should have a back button") {
+                expect(viewController.navigationItem.leftBarButtonItem?.image) == #imageLiteral(resourceName: "arrow_left")
+            }
+        }
+        
+        describe("when child controller did set") {
+            context("if action is add") {
+                beforeEach {
+                    viewController.selectedAddress = nil
+                    _ = viewController.view
+                }
+                
+                it("should have correct child controller properties", closure: {
+                    let childController = viewController.childViewControllers.first as? AddressFormViewController
+                    expect(childController?.address).to(beNil())
+                })
+            }
+            
+            context("if action is edit") {
+                beforeEach {
+                    viewController.selectedAddress = Address()
+                    _ = viewController.view
+                }
+                
+                it("should have correct child controller properties") {
+                    let childController = viewController.childViewControllers.first as? AddressFormViewController
+                    expect(childController?.address) === viewController.selectedAddress
+                }
+            }
+        }
+        
+        describe("when address added") {
+            beforeEach {
+                viewController.addressAction = .add
+                _ = viewController.view
+                viewModelMock.addCustomerAddress(with: Address())
+            }
+            
+            it("should add address") {
+                expect(viewControllerDelegateMock.addressDidAdd) == true
+            }
+        }
+        
+        describe("when address updated") {
+            beforeEach {
+                viewController.addressAction = .edit
+                _ = viewController.view
+                viewModelMock.updateCustomerAddress(with: Address())
+            }
+            
+            it("should update address") {
+                expect(viewControllerDelegateMock.addressDidUpdate) == true
+            }
+        }
+        
+        describe("when filled address did set") {
+            var disposeBag: DisposeBag!
+            var address: Address!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+                address = Address()
+            }
+            
+            it("should return correct address") {
+                viewModelMock.filledAddress.onNext(address)
+                viewController.viewModel.filledAddress
+                    .subscribe(onNext: { filledAddress in
+                        expect(filledAddress) === address
```

Comment:
This test doesn't cover logic from AccountAddressFormViewController class. You need to check that emission of filledAddress calls delegate with returned value